great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py

```diff
@@ -40,17 +49,25 @@ def _build_expectation_configuration(
         parameter_name: str
         fully_qualified_parameter_name: str
         expectation_kwargs: Dict[str, Any] = {
-            parameter_name: get_parameter_value(
-                fully_qualified_parameter_name=fully_qualified_parameter_name,
+            parameter_name: get_parameter_value_and_validate_return_type(
                 domain=domain,
+                parameter_reference=fully_qualified_parameter_name,
+                expected_return_type=None,
                 variables=variables,
                 parameters=parameters,
             )
             for parameter_name, fully_qualified_parameter_name in self._expectation_kwargs.items()
         }
+        meta: Dict[str, Any] = get_parameter_value_and_validate_return_type(
```

Comment:
@jcampbell : For example, let's say that we have defined
```
        false_positive_rate: $variables.false_positive_rate
```
in YAML configuration.  We want to make sure that a floating point value is returned.  It is a valuable safety check.  What if the user above in the "variables" section accidentally specified `false_positive_rate: some_string`?  We want to prevent that.

Thanks.